docs/build-on-btc/btc-dev-env.mdx

```diff
@@ -0,0 +1,309 @@
+---
+
+import TabItem from "@theme/TabItem";
+import { AdornedTabs } from "/src/components/Tabs/AdornedTabs";
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+import { OsType, useOs } from "/src/hooks/useOs";
+
+# BTC developer environment
+
+<MarkdownChipRow labels={["Bitcoin"]} />
+
+To develop Bitcoin applications to be deployed on ICP, your local developer environment will need to include:
+
+- Necessary tools and packages for the language you’d like to build your dapp in:
+
+    - The Rust toolchain for installing Rust packages and compiling Rust code.
+
+    - The Motoko compiler and base library for writing and compiling Motoko code.
+
+- The IC SDK for creating, deploying, and managing smart contracts.
+
+- A local Bitcoin testnet.
+
+- A local instance of the Bitcoin API.
+
+- An ICP smart contract project.
+
+## Install tooling
+
+### Rust toolchain
+
+Before developing BTC applications in Rust, you will need to install the Rust toolchain, including:
+
+- [The Rust programming language.](https://www.rust-lang.org/tools/install)
+- [The `cargo` package manager.](https://doc.rust-lang.org/cargo/getting-started/installation.html)
+
+### Motoko compiler and base library
+
+The Motoko compiler (`moc`) and base library are included in the IC SDK installation as described below. Alternatively, you can install them on their own through the [Motoko release binaries](https://github.com/dfinity/motoko/releases).
+
+### IC SDK
+
+The IC SDK includes a CLI tool called `dfx` that is used for creating, managing, and deploying dapps on ICP. You can install it natively on macOS and Linux; however, Windows users will need to set up WSL 2 before installing the IC SDK.
+
+[Learn more about installing the IC SDK](/docs/building-apps/getting-started/install)
+
+## Create or download an example project
+
+To set up and test your local Bitcoin testnet, you will need a smart contract that implements methods that call the local Bitcoin API.
+
+Create a new project using `dfx new my_project` or download an example in [Motoko](https://github.com/dfinity/examples/tree/master/motoko/basic_bitcoin) or [Rust](https://github.com/dfinity/examples/tree/master/rust/basic_bitcoin) that already implements basic methods for calling the Bitcoin API.
+
+:::tip
+
+Popular libraries like [Rust's bitcoin crate](https://crates.io/crates/bitcoin) and [Motoko's bitcoin package](https://mops.one/bitcoin) can be used within ICP smart contracts.
+
+:::
+
+## Create a local Bitcoin testnet (regtest) with `bitcoind`
+
+It is recommended that developers set up a local Bitcoin testnet on their machine, as it allows them to mine blocks quickly and at will, which facilitates testing various cases without having to rely on the (slow) Bitcoin testnet or the (even slower) Bitcoin mainnet. Alternatively, you can test dapps using the Bitcoin testnet or mainnet through the ICP Bitcoin API. Both workflows are detailed below.
```

Comment:
Bitcoin testnet and mainnet have the same inter-block time, so Bitcoin mainnet is not really slower than testnet.